webapp/src/Components/Replay/ReplayViewer/ReplayViewer.tsx

```diff
@@ -1,27 +1,33 @@
 import {Grid, Typography} from "@material-ui/core"
 import * as React from "react"
-import {RouteComponentProps} from "react-router-dom"
 import {getReplayViewerData} from "../../../Requests/Replay"
 import {ThreeScene} from "./ThreeScene"
+import {Replay} from "../../../Models/Replay/Replay"
+import {ChangeEvent} from "react"
 
 interface OwnProps {
-    id: string
+    replay: Replay
 }
 
-type Props = RouteComponentProps<OwnProps>
+type Props = OwnProps
 
 interface State {
-    replayData?: any
+    replayData?: any,
+    currentFrame: number,
 }
 
 export class ReplayViewer extends React.PureComponent<Props, State> {
     constructor(props: Props) {
         super(props)
-        this.state = {}
+        this.state = {
+            replayData: null,
```

Comment:
Don't set it to null. just delete this line.

Reply:
setting to `null` signals that if you are trying to access `replayData` that the property is valid, but has no data. Otherwise it would be undefined, which is ambiguous and could mean a typo'd property, or no data. IMO I prefer to initialize to null to be more explicit. 
But if preferences prefer just deleting the line, I'll remove it.

Reply:
I prefer deleting the line. Doesn't typescript also complain if you set it to null?
Oh wait it doesn't because it's typed as any :/. 

I'd also like replayData be more specifically typed.